python/tank/util/metrics.py

```diff
@@ -345,79 +348,164 @@ def data(self):
+    event name to the constructor:
+
+    Optionally, you can add your own specific metrics by using the
+    `add_event_property` method. The later simply takes a key-value
+    pair and properly structure the entry into the metric structure.
+
+    metric.add_event_property("MyAppVersion", "10.13.1")
+    metric.add_event_property("ViewedPageCount", 99 )
+
+    Optionally, you can add system information by calling the
+    `add_system_info_properties` method. The method takes care of gathering
+    relevant information and properly formatting it for submission to a
+    Shotgun site.
+
+    Also optionally, you can add user information by calling the
+    `add_user_info_properties`. The method takes care of gathering relevant
+    information regarding location, country, used language and properly
+    formatting it for submission to a Shotgun site.
+
+    Below is a complete typical usage:
+
+    ```
+    metric = EventMetric(event_group="App", event_name="Logged In")
+    metric.add_event_property("MyAppVersion", "10.13.1")
+    metric.add_event_property("ViewedPageCount", 99 )
+    metric.add_system_info_properties()
+    metric.add_user_info_properties()
+    log_event_metric(metric)
+    ```
+
+    TODO: How about event type which is defined in the 'Shotgun Event Taxonomy' document as event property???
+    """
+
+    def __init__(self, event_group, event_name):
+        """
+        Initialize a metric event using the specified parameters. Use this helper class
+        to create a suitable metric structure to be
+
+        :param str event_group: A group or category this metric event falls into.
+            Below are a few Typical values:
+                Trial Signup
+                User Signup
+                App
+                Onboarding
+                Account Settings
+                Projects
+                Entities
+                Tasks
+                People
+                Media
+                Notes
+                Subscribe
+
+
+        :param: str event_name: A short description of the performed action.
+            The complete list can be found in the 'Shotgun Event Taxonomy' document.
+            Below are a few examples:
+                'Viewed Login Page'
+                'Logged In'
+                'Created Project'
+                'Toggled Project Favorite'
+                'Edited Task Status'
+                'Read Inbox Item'
         """
-        super(UserActivityMetric, self).__init__({
-            "type": "user_activity",
-            "module": module,
-            "action": action,
+
+        if event_group:
+            if not isinstance(event_group, str):
+                raise TypeError("The `event_group` parameter must be None or a str")
+
+        if event_name:
+            if not isinstance(event_name, str):
+                raise TypeError("The `event_name` parameter must be None or a str")
+
+        # TODO: go for silent cast to 'str' or raise TypeError ?
```

Comment:
Yes, I would suggest removing the above checks and simply doing `str(event_group)` and `str(event_name)` to avoid raising in this code.